test/modules/jupyter-widget/utils.spec.js

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
```

Comment:
Understand that this is a copied file. Still, worth adding some comments about what these functions are. Imaging the next person who needs to make sense of this file, you have most likely already spent the time understanding what this does.

I am assuming this is a mock connection to a notebook?

Reply:
That's what it is--I'll leave a note.